packages/medusa/src/services/order.js

```diff
@@ -450,6 +455,12 @@ class OrderService extends BaseService {
         )
       }
 
+      for (const item of cart.items) {
+        await this.inventoryService_
+          .withTransaction(manager)
+          .confirmInventory(item.variant_id, item.quantity)
```

Comment:
We bred to do some sort of compensating action on the payment in case this fails. The payment should technically be authorized at this point so if the order creation fails we have payment that is not linked to an order. @olivermrbl What are your thoughts?

Reply:
I agree! Let's make sure to `cancelPayment`, if this stage in cart completion fails @kasperkristensen

Reply:
@olivermrbl, @srindom  Made the requested changes and it now catches the error on no inventory and cancels the payment. Added a unit test to validate the outcome in `/__tests__/order.js` at `fails because an item does not have the required inventory`.